### PR TITLE
Added filter_query to filter builder

### PR DESCRIPTION
**Pull request by praveen5959**
Added `filter_query` to all types of filters.

Testing Scenarios:
1. When the query is present in the query parameter, and the page is refreshed:
Current Behavior: The query is applied but it shows filter type as SQL
Expected Behavior: The query should be applied with appropriate filter type view in the filter section.
2. When a filter is saved:
Current Behavior: filter_builder is being sent in the payload.
Expected Behavior: `filter_builder` will not be sent, instead `filter_query` will have the query.
3. When a filter is saved:
Current Behavior: `filter_builder` is being sent in the payload.
Expected Behavior: `filter_builder` will not be sent, instead `filter_query` will have the query.
4. When a saved filter is applied:
Current Behavior: The filter query  is applied but it always shows the filter view as SQL mode.
Expected Behavior: The filter should be applied as per the filter type and the respective view should be showed in the filter bar.
5. Editing a  saved filter:
Current Behavior: `filter_builder` field is sent as per the `filter_type` in the PUT call
Expected Behavior: `filter_builder` field will not be sent in the PUT call
6. New URL query parameter on filter application:
Expected Behavior: A new URL query parameter should be added to represent `filter_type` when a filter is applied, to track and maintain the filter type within the URL.

### CODE REVIEW

**Review by Koustavd18 — CHANGES_REQUESTED** [review]
filter_query should be not be optional field for createSavedFilterType

File: src/@types/parseable/api/savedFilters.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ export type SavedFilterType = {
 	filter_id: string;
 	query: {
 		filter_type: 'sql' | 'builder';
-		filter_query?: string;
+		filter_query: string;
```
Comment (Koustavd18) [inline src/@types/parseable/api/savedFilters.ts:10]: existing saved filters might not have the filter_query . This type should it have as optional

File: src/pages/Stream/utils.ts [hunk 2]
```diff
@@ -1,17 +1,42 @@
...
+import { parseSQL } from 'react-querybuilder';
+import { QueryType, RuleGroupTypeOverride, RuleTypeOverride } from './providers/FilterProvider';
 
 export const getPageSlice = (page = 1, perPage: number, data: Log[]) => {
 	const firstPageIndex = (page - 1) * perPage;
 	const lastPageIndex = firstPageIndex + perPage;
 	return data ? data.slice(firstPageIndex, lastPageIndex) : [];
 };
 
+export const parseSQLWithIDs = (sqlString: string) => {
```
Comment (Koustavd18) [inline src/pages/Stream/utils.ts:13]: this variable name can be misleading. Please change it to a more appropriate name

File: src/pages/Stream/hooks/useParamsController.ts [hunk 3]
```diff
@@ -54,15 +58,17 @@ const storeToParamsObj = (opts: {
...
-	const { timeRange, offset, page, view, rows, query } = opts;
+	const { timeRange, offset, page, view, rows, query, filterType } = opts;
 	const params: Record<string, string> = {
 		...deriveTimeRangeParams(timeRange),
 		view,
 		offset,
 		rows,
 		page,
 		query,
+		filterType,
```
Comment (Koustavd18) [inline src/pages/Stream/hooks/useParamsController.ts:71]: we can add a condition here to set the filter type in the url params when there is a valid query

File: src/pages/Stream/utils.ts [hunk 4]
```diff
@@ -1,17 +1,42 @@
...
+
+	addIds(parsedQuery);
+	return parsedQuery;
+};
+
 export const makeHeadersFromSchema = (schema: LogStreamSchemaData | null): string[] => {
 	if (schema) {
-		const { fields } = schema;
-		return fields.map((field) => field.name);
+	const { fields } = schema;
```
Comment (Koustavd18) [inline src/pages/Stream/utils.ts:38]: code not properly formatted here

**Review by Koustavd18 — CHANGES_REQUESTED** [review]

File: src/pages/Stream/utils.ts (hunk 4 above)
Comment (praveen5959) [reply src/pages/Stream/utils.ts:38]: Addressed this

File: src/pages/Stream/utils.ts (hunk 2 above)
Comment (praveen5959) [reply src/pages/Stream/utils.ts:13]: Addressed this

File: src/hooks/useSavedFilters.tsx [hunk 5]
```diff
@@ -26,7 +26,10 @@ const useSavedFiltersQuery = () => {
 	);
 
 	const { mutate: updateSavedFilters, isLoading: isUpdating } = useMutation(
-		(data: { filter: SavedFilterType; onSuccess?: () => void }) => putSavedFilters(data.filter.filter_id, data.filter),
+		(data: { filter: SavedFilterType; onSuccess?: () => void }) => {
+			data.filter.query = _.omit(data.filter.query, 'filter_builder');
```
Comment (Koustavd18) [inline src/hooks/useSavedFilters.tsx:30]: we should do this only when filter builder exists.
Also add a comment why this line is added

File: src/pages/Stream/components/Querier/FilterQueryBuilder.tsx [hunk 6]
```diff
@@ -116,18 +116,18 @@ const CombinatorToggle = (props: CombinatorToggleType) => {
...
-				className={isOrSelected ? activeBtnClass : inActiveBtnClass}
-				onClick={() => onCombinatorChange('or')}>
-				OR
-			</Text>
 			<Text
 				style={{ fontSize: '0.6rem' }}
 				className={!isOrSelected ? activeBtnClass : inActiveBtnClass}
 				onClick={() => onCombinatorChange('and')}>
 				AND
 			</Text>
```
Comment (Koustavd18) [inline src/pages/Stream/components/Querier/FilterQueryBuilder.tsx:124]: OR is still the default combinator

**Review by Koustavd18 — CHANGES_REQUESTED** [review]

File: src/hooks/useSavedFilters.tsx (hunk 5 above)
Comment (praveen5959) [reply src/hooks/useSavedFilters.tsx:30]: Addressed this.

**Review by Koustavd18 — DISMISSED** [review]

**Review by nikhilsinhaparseable — APPROVED** [review]
good to merge

**Review by Koustavd18 — APPROVED** [review]

**Pull request merged**